src/drive/ducks/files/Toolbar.jsx

```diff
@@ -58,146 +97,97 @@ class Toolbar extends Component {
+          </ToolbarAction>
+          <ToolbarAction
+            visible={actions.addFolder && hasWriteAccess}
+            highlighted={false}
           >
-            {t('toolbar.menu_select')}
-          </a>
-        </Item>
-        {shared.withMe && <hr />}
-        {shared.withMe && (
-          <Item>
             <a
-              className={classNames(styles['fil-action-delete'])}
+              className={styles['fil-action-newfolder']}
+              onClick={actions.addFolder}
+            >
+              {t('toolbar.menu_new_folder')}
+            </a>
+          </ToolbarAction>
+          <ToolbarAction visible highlighted={false}>
+            <a
+              className={styles['fil-action-select']}
+              onClick={onSelectItemsClick}
+            >
+              {t('toolbar.menu_select')}
+            </a>
+          </ToolbarAction>
+          <ToolbarAction visible={shared.withMe} highlighted={false}>
+            <hr />
+          </ToolbarAction>
+          <ToolbarAction visible={shared.withMe} highlighted={false}>
+            <a
+              className={styles['fil-action-delete']}
               onClick={() => leaveFolder(displayedFolder)}
             >
               {t('toolbar.leave')}
             </a>
-          </Item>
-        )}
-      </Menu>
-    )
-
-    return (
-      <div className={styles['fil-toolbar-files']} role="toolbar">
-        {!shared.withMe &&
-          (cozyDev || cozyRecette
-            ? (console.warn('IntentButton is displayed only on dev or recette'),
-              (
-                <IntentButton
-                  className={classNames(styles['c-btn'], styles['u-hide--mob'])}
-                  action="CREATE"
-                  docType="io.cozy.accounts"
-                  data={{
-                    dataType: 'bill',
-                    closeable: false
-                  }}
-                >
-                  {t('service.bills')}
-                </IntentButton>
-              ))
-            : null)}
-        {hasWriteAccess && (
-          <UploadButton
-            disabled={disabled}
-            onUpload={files => uploadFiles(files, displayedFolder)}
-            label={t('toolbar.item_upload')}
-            className={classNames(styles['c-btn'], styles['u-hide--mob'])}
-          />
-        )}
-        {notRootfolder &&
-          !shared.withMe && (
-            <ShareButton
-              disabled={disabled}
-              onClick={() => this.setState(toggleShowShareModal)}
-              label={t('toolbar.share')}
-              className={styles['u-hide--mob']}
-            />
-          )}
-        {shared.withMe && (
-          <SharedWithMeButton
-            label={t('Files.share.sharedWithMe')}
-            onClick={() => this.setState(toggleShowShareModal)}
-            className={styles['u-hide--mob']}
-          />
-        )}
-
-        {isMobile ? <BarRight>{MoreMenu}</BarRight> : MoreMenu}
+          </ToolbarAction>
+        </Toolbar>
```

Comment:
This very long `<Toolbar><ToolbarAction /></Toolbar>` is very hard to read.
Can't we find a way to refactor and minimize the code to clarify its reading?

Reply:
Any suggestions? I think `Toolbar` is quite good, and the sub-component should be `ToolbarSomething`. i'm not too happy with `Action` though.

Reply:
Without giving it too much time, I will go with a configuration solution:

1. create an object or a set of value
1. loop on this objet to create different items